application/controllers/collection_instrument.py

```diff
@@ -87,42 +88,50 @@ def upload_to_bucket(self, exercise_id, file, ru_ref=None, classifiers=None, ses
         :param session: database session
         :return: a collection instrument instance
         """
-
-        log.info("Upload exercise", exercise_id=exercise_id)
+        log.info("Upload instrument", exercise_id=exercise_id)
 
         validate_uuid(exercise_id)
         self.validate_non_duplicate_instrument(file, exercise_id, session)
-        instrument = InstrumentModel(ci_type="SEFT")
-
-        seft_file = self._create_seft_file(instrument.instrument_id, file)
-        instrument.seft_file = seft_file
-
-        exercise = self._find_or_create_exercise(exercise_id, session)
-        instrument.exercises.append(exercise)
 
         survey = self._find_or_create_survey_from_exercise_id(exercise_id, session)
-        instrument.survey = survey
+        survey_id = survey.survey_id
+        survey_service_details = get_survey_service_details(survey_id)
```

Comment:
I moved the order around as once past validate_uuid and validate_non_duplicate_instrument the next place you could fall over is the call to the survey service and you can't avoid it. Therefore it makes sense to call it next to save yourself doing a load of stuff that will just blow up later if it fails